app/main.py

```diff
@@ -1,3 +1,86 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+import datetime
+from app.car import Car
+from app.customer import Customer
+from app.shop import Shop
+
+with open("app/config.json") as config:
+    data = json.load(config)
+
+customers = []
+for customer_data in data["customers"]:
+    car_data = customer_data.pop("car")
+    car = Car(**car_data)
+    customer = Customer(car=car, **customer_data)
+    customers.append(customer)
+
+shops = [Shop(**shop_data) for shop_data in data["shops"]]
+
+fuel_price = data["FUEL_PRICE"]
+
+
+def find_nearest_shop(customer: Customer,
+                      shops: list,
+                      fuel_price: float) -> tuple:
+    min_cost = float("inf")
+    nearest_shop = None
+    costs = []
+    for shop in shops:
+        trip_cost = customer.cost_to_shop(shop.location, fuel_price, shop)
+        costs.append((shop.name, trip_cost))
+        if trip_cost < min_cost:
+            min_cost = trip_cost
+            nearest_shop = shop
+    return nearest_shop, min_cost, costs
+
+
+def shop_trip() -> None:
+    for customer in customers:
+        nearest_shop, min_cost, costs = (
+            find_nearest_shop(customer, shops, fuel_price))
+        print(f"{customer.name} has {customer.money} dollars")
+        for shop_name, trip_cost in costs:
+            print(f"{customer.name}'s trip to the "
+                  f"{shop_name} costs {trip_cost:.2f}")
+
+        product_cost = 0
+        for product, quantity in customer.product_cart.items():
+            product_cost += nearest_shop.products[product] * quantity
+
+        if customer.money < min_cost + product_cost:
+            print(
+                f"{customer.name} doesn't have enough money "
+                f"to make a purchase in any shop"
+            )
+            continue
+
+        print(f"{customer.name} rides to {nearest_shop.name}")
```

Comment:
when customer rides home his location should be shop location, the same with home location